### PR TITLE
Pim 5853 ope

### PR DESCRIPTION
**Pull request by willy-ahva**
<3 Thanks for taking the time to contribute! You're awesome! <3

If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/CONTRIBUTING.md 

**Description (for Contributor and Core Developer)**

What does this Pull Request do? reference the related issue?

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       |
| Added Behats                      |
| Changelog updated                 |
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+     * @param array  $options ['lastJobExecution' => JobExecution]
+     */
+    protected function addUpdatedSinceLastExport($field, $options = [])
+    {
+        $this->checkOptions($field, $options);
+
+        if (null !== $options['lastJobExecution']) {
+            $lastJobEndTime = $options['lastJobExecution']->getEndTime();
+
+            // TODO : "Greater than" or "Greater or equal than" ??
```
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: Need to know

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php [hunk 2]
```diff
@@ -46,24 +46,43 @@ public function __construct(
...
     public function addFieldFilter($field, $operator, $value, $locale = null, $scope = null, $options = [])
     {
-        $this->checkValue($field, $value, $locale, $scope);
+        $expr = $this->qb->expr();
 
-        $localeCodes = (null !== $locale) ?
-            [$locale] :
-            $this->getChannelByCode($scope)->getLocaleCodes();
+        switch ($operator) {
+            case Operators::COMPLETE_ON_ALL_LOCALES:
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: we should pass the value

Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: It should be MORE_THAN_ON_LOCALE

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php [hunk 3]
```diff
@@ -46,24 +46,43 @@ public function __construct(
...
+                $this->checkScope($field, $scope);
+
+                foreach ($options['locales'] as $localeCode) {
+                    $field = $this->getNormalizedField($scope, $localeCode);
+                    $expr->addAnd($this->getExpr(100, $field, Operators::EQUALS));
+                }
+                break;
 
-        $expr = $this->qb->expr();
+            case Operators::NOT_COMPLETE_ON_ALL_LOCALES:
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: LOWER_THAN_ON_LOCALES

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addFieldFilter($field, $operator, $value, $locale = null, $scope = null, $options = [])
+    {
+        $field = sprintf('%s.%s', ProductQueryUtility::NORMALIZED_FIELD, $field);
+
+        if (Operators::SINCE_LAST_EXPORT === $operator) {
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: SINCE_LAST_JOB_EXECUTION. it should not bit tighten  with export

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+     *
+     * @param string $field
+     * @param array  $options ['lastJobExecution' => JobExecution]
+     */
+    protected function addUpdatedSinceLastExport($field, $options = [])
+    {
+        $this->checkOptions($field, $options);
+
+        if (null !== $options['lastJobExecution']) {
+            $lastJobEndTime = $options['lastJobExecution']->getEndTime();
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: we should pass it as a value, not context.

Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: We should pass the code of the job

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php (hunk 2 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: MORE THAN ON ALL LOCALES

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php [hunk 6]
```diff
@@ -46,24 +46,43 @@ public function __construct(
...
-            [$locale] :
-            $this->getChannelByCode($scope)->getLocaleCodes();
+        switch ($operator) {
+            case Operators::COMPLETE_ON_ALL_LOCALES:
+                $this->checkOptions($field, $options);
+                $this->checkScope($field, $scope);
+
+                foreach ($options['locales'] as $localeCode) {
+                    $field = $this->getNormalizedField($scope, $localeCode);
+                    $expr->addAnd($this->getExpr(100, $field, Operators::EQUALS));
```
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: operator sup or equal

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php (hunk 3 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: LOWER_OR_EQUAL_THAN_ON_ALL_LOCALES

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php [hunk 7]
```diff
@@ -46,24 +46,43 @@ public function __construct(
      */
     public function addFieldFilter($field, $operator, $value, $locale = null, $scope = null, $options = [])
     {
-        $this->checkValue($field, $value, $locale, $scope);
+        $expr = $this->qb->expr();
```
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: Do 4 operator lte lt gt gte

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php (hunk 4 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: $value = lastJobCode
Repo->getLastJob

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php (hunk 5 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: getstartedTime

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php (hunk 1 above)
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: gt

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/CompletenessFilter.php [hunk 8]
```diff
@@ -39,23 +39,26 @@ public function addFieldFilter($field, $operator, $value, $locale = null, $scope
         switch ($operator) {
             case Operators::COMPLETE_ON_ALL_LOCALES:
                 $this->checkOptions($field, $options);
+                $this->checkScope($field, $scope);
 
                 foreach ($options['locales'] as $locale) {
-                    $this->addFilter(Operators::EQUALS, '100', $locale, $scope);
+                    $this->applyFilter(Operators::EQUALS, '100', $locale, $scope);
```
Comment (willy-ahva) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/CompletenessFilter.php]: do a mapping between opérators

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php [hunk 9]
```diff
@@ -46,24 +54,27 @@ public function __construct(
...
+                $expr->addAnd($this->getExpr($value, $field, $this->operatorsMapping[$operator]));
+            }
+        } else {
+            $localeCodes = (null !== $locale) ?
+                [$locale] :
+                $this->getChannelByCode($scope)->getLocaleCodes();
+
+            foreach ($localeCodes as $localeCode) {
+                $field = $this->getNormalizedField($scope, $localeCode);
+                $value = intval($value);
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: please use a proper cast

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+     * @param string $operator
+     * @param mixed  $value
+     */
+    protected function checkValue($field, $operator, $value)
+    {
+        if ($operator === Operators::SINCE_LAST_EXPORT && !is_string($value)) {
+            throw InvalidArgumentException::stringExpected($field, 'filter', 'updated', gettype($value));
+        }
+
+        if ($operator !== Operators::SINCE_LAST_EXPORT && !is_numeric($value)) {
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: why?

Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/UpdatedDateTimeFilter.php]: you mean $operator === Operators::SINCE_LAST_EXPORT right?

File: src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/UpdatedDateTimeFilter.php [hunk 11]
```diff
@@ -55,61 +71,58 @@ protected function addSinceLastNDays($field, $value)
...
-                $field,
-                sprintf(
-                    'option "lastJobExecution" to be an instance of "%s" or null value',
-                    JobExecution::class
-                ),
-                'filter',
-                'updated datetime',
-                print_r($options['lastJobExecution'], true)
-            );
+        if ($operator !== Operators::SINCE_LAST_EXPORT && !is_numeric($value)) {
```
Comment (juliensnz) [inline src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Filter/UpdatedDateTimeFilter.php]: same

File: src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php (hunk 9 above)
Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/Doctrine/MongoDBODM/Filter/CompletenessFilter.php]: `(int)`

File: src/Pim/Bundle/ImportExportBundle/JobParameters/FormConfigurationProvider/ProductCsvExport.php [hunk 12]
```diff
@@ -47,26 +41,20 @@ class ProductCsvExport implements FormConfigurationProviderInterface
...
         array $dateFormats
     ) {
-        $this->simpleCsvExport = $simpleCsvExport;
-        $this->familyRepository  = $familyRepository;
-        $this->channelRepository = $channelRepository;
+        $this->simpleCsvExport   = $simpleCsvExport;
         $this->supportedJobNames = $supportedJobNames;
         $this->decimalSeparators = $decimalSeparators;
-        $this->dateFormats = $dateFormats;
+        $this->dateFormats      = $dateFormats;
```
Comment (phaseinducer) [inline src/Pim/Bundle/ImportExportBundle/JobParameters/FormConfigurationProvider/ProductCsvExport.php]: :scream:

**Pull request merged**